exporter/azuremonitorexporter/logexporter_test.go

```diff
@@ -128,6 +129,50 @@ func TestExporterLogDataCallback(t *testing.T) {
 	mockTransportChannel.AssertNumberOfCalls(t, "Send", 3)
 }
 
+func TestLogRecordToEnvelopeResourceAttributes(t *testing.T) {
+	logRecord := getTestLogRecord(t, 1)
+	logPacker := getLogPacker()
+	resource := getResource()
+
+	envelope := logPacker.LogRecordToEnvelope(logRecord, resource, getScope())
+
+	require.NotEmpty(t, resource.Attributes())
+	envelopeData := envelope.Data.(*contracts.Data).BaseData.(*contracts.MessageData)
+	require.Subset(t, envelopeData.Properties, resource.Attributes().AsRaw())
+}
+
+func TestLogRecordToEnvelopeInstrumentationScope(t *testing.T) {
+	const aiInstrumentationLibraryNameConvention = "instrumentationlibrary.name"
+	const aiInstrumentationLibraryVersionConvention = "instrumentationlibrary.version"
+
+	logRecord := getTestLogRecord(t, 1)
+	logPacker := getLogPacker()
+	scope := getScope()
+
+	envelope := logPacker.LogRecordToEnvelope(logRecord, getResource(), scope)
+
+	envelopeData := envelope.Data.(*contracts.Data).BaseData.(*contracts.MessageData)
+	require.Equal(t, scope.Name(), envelopeData.Properties[aiInstrumentationLibraryNameConvention])
+	require.Equal(t, scope.Version(), envelopeData.Properties[aiInstrumentationLibraryVersionConvention])
```

Comment:
I think it would make sense to use the same attribute constants that the production code uses here.

Reply:
To make sure I understand the suggestion correctly: the suggestion would be to import the constant `instrumentationLibraryVersion` from `contract_utils.go`, correct?

My thinking behind "duplicating" the constants is that if someone changes the value of `instrumentationLibraryVersion` in the production code, which is a breaking change, there will be a test that breaks.

Hence I would tend to keep it as it is, but no strong opinion here and happy to apply your suggestion if you're not convinced.

Reply:
That's a fair point. I don't feel strongly about it either, so I'm good leaving it as-is.